playground/HealthChecks/HealthChecksSandbox.AppHost/Program.cs

```diff
@@ -24,28 +25,22 @@
 
 builder.Build().Run();
 
-static string GetException(string message)
+IResourceBuilder<TestResource> AddTestResource(string name, HealthStatus status, string? description = null, string? exceptionMessage = null)
 {
-    try
-    {
-        throw new InvalidOperationException(message);
-    }
-    catch (InvalidOperationException ex)
-    {
-        return ex.ToString();
-    }
-}
+    builder.Services.AddHealthChecks()
+                    .AddCheck(
+                        $"{name}_check",
+                        () => new HealthCheckResult(status, description, new InvalidOperationException(exceptionMessage))
+                        );
 
-IResourceBuilder<TestResource> AddTestResource(string name, HealthStatus status, string? description = null, string? exception = null)
-{
     return builder
         .AddResource(new TestResource(name))
+        .WithHealthCheck($"{name}_check")
         .WithInitialState(new()
         {
             ResourceType = "Test Resource",
             State = "Starting",
             Properties = [],
-            HealthReports = [new HealthReportSnapshot($"{name}_check", status, description, exception)]
```

Comment:
Why switch to the annotation? I'm not against the sandbox having health checks driven by the annotation - it's the default experience after all - but it's useful to have some resources set health reports in a snapshot manually to double check that scenario isn't broken.

Reply:
We don’t care about this for 9. We cns re-evaluate if it becomes a real scenario

Reply:
Fair. But, sandboxes are a place to try out dumb stuff. I'd like to keep some examples that manually set health reports just to see what happens if someone does do it.

Reply:
`HealthReports` init is now internal. Users cannot set health reports anymore in snapshots

Reply:
Oh ok. Never mind then.